core/ignore.js

```diff
@@ -1,164 +1,110 @@
+    noslash = false
+  }
+  if (line[0] === '!') {
+    // Detect bang prefix
+    line = line.slice(1)
+    negate = true
+  }
+  if (line[0] === '/') {
+    // Detect leading slash
+    line = line.slice(1)
+  }
+  if (line[line.length - 1] === '/') {
+    // Detect trailing slash
+    line = line.slice(0, line.length - 1)
+    folder = true
+  }
+  line = line.replace(/^\\/, '') // Remove leading escaping char
+  line = line.replace(/( |\t)*$/, '') // Remove trailing spaces
+  // Ignore case for case insensitive file-systems
+  if (process.platform === 'darwin' || process.platform === 'win32') {
+    line = makeRe(line, { nocase: true })
+  }
+  let pattern = {
+    match: matcher(line, {}),
+    basename: noslash, // The pattern can match only the basename
+    folder, // The pattern will only match a folder
+    negate // The pattern is negated
+  }
+  return pattern
+}
 
-/*::
-export type IgnorePattern = {
-  match: (string) => boolean,
-  basename: boolean,
-  folder: boolean,
-  negate: boolean
+function isNotBlankOrComment (line) {
+  return line !== '' && line[0] !== '#'
+}
+
+function match (path, isFolder, pattern) {
+  if (pattern.basename) {
+    if (pattern.match(basename(path))) {
+      return true
+    }
+  }
+  if (isFolder || !pattern.folder) {
+    if (pattern.match(path)) {
+      return true
+    }
+  }
+  let parent = dirname(path)
+  if (parent === '.') {
+    return false
+  }
+  return match(parent, true, pattern)
 }
-*/
 
 // Cozy-desktop can ignore some files and folders from a list of patterns in the
 // cozyignore file. This class can be used to know if a file/folder is ignored.
 //
 // See https://git-scm.com/docs/gitignore/#_pattern_format
-let DefaultRules
-let MicromatchOptions
 class Ignore {
-  static initClass () {
-    // See https://github.com/github/gitignore/tree/master/Global
-    DefaultRules = [
-      // all hidden files
-      '.*',
-
-      // Dropbox
-      '.dropbox',
-      '.dropbox.attr',
-      '.dropbox.cache',
-
-      // Eclipse, SublimeText and many others
-      '*.tmp',
-      '*.bak',
-
-      // Emacs
-      '*~',
-      '\\#*\\#',
-
-      // LibreOffice
-      '.~lock.*#',
-
-      // Linux
-      '.fuse_hidden*',
-      '.Trash-*',
-
-      // Microsoft Office
-      '~$*.{doc,xls,ppt}*',
-
-      // OSX
-      '.DS_Store',
-      '.DocumentRevisions-V100',
-      '.fseventsd',
-      '.Spotlight-V100',
-      '.TemporaryItems',
-      '.Trashes',
-      '.VolumeIcon.icns',
-      // Pattern must be escaped twice on case-insensitive plateforms in order
-      // for makeRe() to work
-      process.platform === 'darwin' || process.platform === 'win32'
-        ? 'Icon\\r'
-        : 'Icon\r',
-
-      // Vim
-      '*.sw[px]',
-
-      // Windows
-      'Thumbs.db',
-      'ehthumbs.db'
-    ]
-
-    // See https://github.com/jonschlinkert/micromatch#options
-    MicromatchOptions =
-            {noextglob: true}
-  }
-
-  /*::
-  patterns: IgnorePattern[]
```

Comment:
These comments `/*::` are flow annotation and the reason why this PR is falling CI (https://travis-ci.org/cozy-labs/cozy-desktop/jobs/446757368#L712)

Run `yarn lint:flow` locally to see the errors.

Reply:
oops, I'll check that.